src/modules/Writers.jl

```diff
@@ -59,13 +59,91 @@ function render(io::IO, mime::MIME"text/plain", node::Expanders.DocsNodes, page,
     end
 end
 
+"""
+Converts the function argument tuple `(name, type)` into a string.
+
+The tuple comes from the second return element of the `Base.arg_decl_parts(::Method)`
+and it seems they are always both `::String` (`::ASCIIString` in 0.4).
+It also appears that if the type is not declared for the method, `arg_decl_parts`
+returns an empty string.
+
+The returned string is `name::type` or just `name`, if the type is not declared.
+"""
+function join_decl(decl)
+    n, t = decl
+    isempty(t) ? n : "$(n)::$(t)"
+end
+
 function render(io::IO, mime::MIME"text/plain", node::Expanders.DocsNode, page, doc)
     # Docstring header based on the name of the binding and it's category.
     anchor = "<a id='$(node.anchor.id)' href='#$(node.anchor.id)'>#</a>"
     header = "**`$(node.object.binding)`** &mdash; *$(Utilities.doccat(node.object))*."
     println(io, anchor, "\n", header, "\n\n")
     # Body. May contain several concatenated docstrings.
     renderdoc(io, mime, node.docstr, page, doc)
+
+    # Table of methods.
+    # If DocsNode.methods is nulled, then we assume that we should not render a
+    # table. However, if the list of methods is there but has no elements then
+    # we output an appropriate note saying that the name has no methods associated
+    # with it.
+    Utilities.unwrap(node.methods) do ms
```

Comment:
Might be worth factoring this out since it's quite big. Then just calling something like

```julia
Utilities.unwrap(ms -> rendermethods(io, node.object, ms), node.methods)
```

if that's possible. Would also help to avoid vars leaking into the closure unintentionally.